server/src/test/java/org/opensearch/index/store/CompositeDirectoryTests.java

```diff
@@ -0,0 +1,189 @@
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.opensearch.index.store.remote.file.CleanerDaemonThreadLeakFilter;
+import org.opensearch.index.store.remote.file.OnDemandBlockSnapshotIndexInput;
+import org.opensearch.index.store.remote.filecache.CachedIndexInput;
+import org.opensearch.index.store.remote.filecache.FileCache;
+import org.opensearch.index.store.remote.filecache.FullFileCachedIndexInput;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.startsWith;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ThreadLeakFilters(filters = CleanerDaemonThreadLeakFilter.class)
+public class CompositeDirectoryTests extends BaseRemoteSegmentStoreDirectoryTests {
+    private FileCache fileCache;
+    private FSDirectory localDirectory;
+    private CompositeDirectory compositeDirectory;
+
+    @Before
+    public void setup() throws IOException {
+        setupRemoteSegmentStoreDirectory();
+        localDirectory = mock(FSDirectory.class);
+        fileCache = mock(FileCache.class);
+        compositeDirectory = new CompositeDirectory(localDirectory, remoteSegmentStoreDirectory, fileCache);
+    }
+
+    public void testListAll() throws IOException {
+        when(localDirectory.listAll()).thenReturn(new String[]{});
+        String[] actualFileNames = compositeDirectory.listAll();
+        String[] expectedFileNames = new String[] {};
+        assertArrayEquals(expectedFileNames, actualFileNames);
+
+        populateMetadata();
+        when(localDirectory.listAll()).thenReturn(new String[] { "_1.cfe", "_2.cfe", "_0.cfe_block_7", "_0.cfs_block_7" });
+
+        actualFileNames = compositeDirectory.listAll();
+        expectedFileNames = new String[] { "_0.cfe", "_0.cfs", "_0.si", "_1.cfe", "_2.cfe", "segments_1" };
+        assertArrayEquals(expectedFileNames, actualFileNames);
+    }
+
+    public void testDeleteFile() throws IOException {
+        Path basePath = mock(Path.class);
+        Path resolvedPath = mock(Path.class);
+        when(basePath.resolve(anyString())).thenReturn(resolvedPath);
+        when(localDirectory.getDirectory()).thenReturn(basePath);
+
+        compositeDirectory.deleteFile("_0.tmp");
+        verify(localDirectory).deleteFile("_0.tmp");
+
+        compositeDirectory.deleteFile("_0.si");
+        verify(fileCache).remove(resolvedPath);
+    }
+
+    public void testFileLength() throws IOException {
+        populateMetadata();
+        remoteSegmentStoreDirectory.init();
+        Path basePath = mock(Path.class);
```

Comment:
I think we should use real paths here with real files.  We also need to test cases where the file does not exist.